hibernate-reactive-core/src/main/java/org/hibernate/reactive/provider/service/AbstractReactiveInformationSchemaBasedExtractorImpl.java

```diff
@@ -0,0 +1,260 @@
+		return normalizeLabel( super.getResultSetIndexNameLabel() );
+	}
+
+	@Override
+	protected String getResultSetForeignKeyLabel() {
+		return normalizeLabel( super.getResultSetForeignKeyLabel() );
+	}
+
+	@Override
+	protected String getResultSetPrimaryKeyNameLabel() {
+		return normalizeLabel( super.getResultSetPrimaryKeyNameLabel() );
+	}
+
+	@Override
+	protected String getResultSetColumnPositionColumn() {
+		return normalizeLabel( super.getResultSetColumnPositionColumn() );
+	}
+
+	@Override
+	protected String getResultSetPrimaryKeyColumnNameLabel() {
+		return normalizeLabel( super.getResultSetPrimaryKeyColumnNameLabel() );
+	}
+
+	@Override
+	protected String getResultSetForeignKeyColumnNameLabel() {
+		return normalizeLabel( super.getResultSetForeignKeyColumnNameLabel() );
+	}
+
+	@Override
+	protected String getResultSetPrimaryKeyCatalogLabel() {
+		return normalizeLabel( super.getResultSetPrimaryKeyCatalogLabel() );
+	}
+
+	@Override
+	protected String getResultSetPrimaryKeySchemaLabel() {
+		return normalizeLabel( super.getResultSetPrimaryKeySchemaLabel() );
+	}
+
+	@Override
+	protected String getResultSetPrimaryKeyTableLabel() {
+		return normalizeLabel( super.getResultSetPrimaryKeyTableLabel() );
+	}
+
+	private String normalizeLabel(String columnLabel) {
+		return toMetaDataObjectName( Identifier.toIdentifier( columnLabel ) );
+	}
+
+	@Override
+	protected <T> T processCatalogsResultSet(ExtractionContext.ResultSetProcessor<T> processor) throws SQLException {
+		return getExtractionContext().getQueryResults(
+				String.format(
+						"select catalog_name as %s from information_schema.information_schema_catalog_name",
+						getResultSetCatalogLabel()
+				),
+				null,
+				processor
+		);
+	}
+
+	@Override
+	protected <T> T processSchemaResultSet(
+			String catalog,
+			String schemaPattern,
+			ExtractionContext.ResultSetProcessor<T> processor) throws SQLException {
+		final StringBuilder sb = new StringBuilder()
+				.append( "select catalog_name as " ).append( getResultSetCatalogLabel() )
+				.append( " , schema_name as "  ).append( getResultSetSchemaLabel() )
+				.append( " from information_schema.schemata where true" );
+		final List<Object> parameters = new ArrayList<>();
+		appendClauseAndParameterIfNotNull( " and catalog_name = ", catalog, sb, parameters );
+		appendClauseAndParameterIfNotNull( " and schema_name like ", schemaPattern, sb, parameters );
+		return getExtractionContext().getQueryResults( sb.toString(), parameters.toArray(), processor );
+	}
+
+	protected void appendClauseAndParameterIfNotNull(
+			String clause,
+			Object parameter,
+			StringBuilder sb,
+			List<Object> parameters) {
+
+		if ( parameter != null ) {
+			parameters.add( parameter );
+			sb.append( clause );
+			// TODO: PostgreSQL does not seem to accept "?" parameter placeholders.
+			//       It does accept $<position>. Will this work for all dialects?
+			sb.append( "$").append( parameters.size() );
```

Comment:
I see that this doesn't work for MySQL. It looks like `PostgresParameters` knows to make this change.

I'm looking into how I can change this to use `Parameters`...

Reply:
I just added a [commit](https://github.com/hibernate/hibernate-reactive/pull/903/commits/cd964b9b5609045249954280779c117e85d2d452) to use `Parameters` to get the dialect-specific placeholder.